src/uri/mod.rs

```diff
@@ -110,6 +110,9 @@ pub struct Parts {
     /// The authority component of a URI
     pub authority: Option<Authority>,
 
+    /// The port component of a URI
+    pub port: Option<u16>,
```

Comment:
The port is part of `authority`, we probably shouldn't have it in a two places.

Reply:
I am care about different order: `.port().authority()`,
but looks like `Uri::Builder` do not handle the misuse like `.authority().authority()`,
and so on. So I made changes as you suggest, so in new variant 
`.port().authority()` authority just overwrite what `port` is done.